src/LaunchDarkly.ServerSdk/InMemoryFeatureStore.cs

```diff
@@ -31,9 +30,9 @@ public class InMemoryFeatureStore : IFeatureStore
         /// <inheritdoc/>
         public T Get<T>(VersionedDataKind<T> kind, string key) where T : class, IVersionedData
         {
+            RwLock.EnterReadLock();
             try
             {
-                RwLock.TryEnterReadLock(RwLockMaxWaitMillis);

```

Comment:
Because you are not checking the result of the `TryEnterReadLock`, which indicates whether or not you got the lock, this class is not thread safe.

One thread can acquire the lock, and another thread can come along, wait for a second, then begin mutating the same dictionaries.

Reply:
See documentation on how to use:

https://docs.microsoft.com/en-us/dotnet/api/system.threading.readerwriterlockslim?view=netframework-4.8